lib/constrainer.js

```diff
@@ -153,10 +153,8 @@ class Constrainer {
       if (!strategy.isCustom) {
         if (key === 'version') {
           lines.push('   version: req.headers[\'accept-version\'],')
-        } else if (key === 'host') {
-          lines.push('   host: req.headers.host || req.headers[\':authority\'],')
         } else {
-          throw new Error('unknown non-custom strategy for compiling constraint derivation function')
+          lines.push('   host: req.headers.host || req.headers[\':authority\'],')
```

Comment:
Uncovered branch.

How can we accidentally inject a builtin strategy or forget to manage it?